docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3996,9 +4026,18 @@ A `WriteRequest` may only include an `Update` message of type `MODIFY` with a
   specified by the client. Note that if this Protobuf field is not set, the
   meter config is set to execute the default behavior (GREEN for all packets).
 
+* `counter_data` is used to set the per color counter values for the
+   corresponding meter entry to the default value of (0), which is
+   essentially clearing the counters. Targets not supporting this counter should
+   return `UNIMPLEMENTED` if any or all of the subfields are set. Note that if
+   this Protobuf field or any of the specific color <color> field is not set,
+   the counter value is not changed.
+
 Specifying `DirectMeterEntry` in an `Update` message of type `INSERT` or
 `DELETE` is not allowed, and the server must return the error code
-`INVALID_ARGUMENT` in that case.
+`INVALID_ARGUMENT` in that case. Targets not supporting per color meter
+counters can reject a write request that has this field set with an
+`UNIMPLEMENTED` error code.
```

Comment:
The earlier paragraph is for MODIFY and this note is for INSERT or DELETE.

Reply:
This does not make sense to me. If you have an update of type `INSERT` or `DELETE` such that `update.entity().direct_meter_entry().has_counter_data()` is true, then this implies that `update.entity().has_direct_meter_entry()` is true, which means we must return `INVALID_ARGUMENT` by the first sentence in this paragraph.

Reply:
> can reject a write request that has this field set with an `UNIMPLEMENTED` error code.

Can we make this unambiguous, "can reject" -> "must reject"?

On a related note, maybe it would be better to state a prior that any `DirectMeterEntry` message with `counter_data` set must be rejected with UNIMPLEMENTED by implementations that do not support it. That way we don't have to repeat this again and again for the various types of requests, and there is no ambiguity.

Reply:
Updated the unimplemented part in the 'counter_data' field text and removed the repetitions.